util/network-devp2p/src/host.rs

```diff
@@ -475,7 +543,17 @@ impl Host {
 			Some(addr) => NodeEndpoint { address: addr, udp_port: local_endpoint.udp_port }
 		};
 
-		self.info.write().public_endpoint = Some(public_endpoint.clone());
+		(|| {
```

Comment:
I don't think the closure is needed, just scoping the code should have the same effect:
```rust
{
  ...
}
```